### PR TITLE
Implemented Feature: EventCategory plain enum To enum class.

### PR DESCRIPTION
**Pull request by babaliaris**
#### Issue Description (This is not related to any opened issues).
The plain enum **EventCategory** was causing a *warning* because plain enums are globally scoped!
enum class should be always used instead of plain enums. You can read more about it here:
https://stackoverflow.com/questions/18335861/why-is-enum-class-preferred-over-plain-enum

#### How did I fix it:
- [x] I changed the **EventCategory** plain enum to an **enum class**.
- [x] In order for the **bitwise operations** to work 
```
EVENT_CLASS_CATEGORY(EventCategory::EventCategoryMouse | EventCategory::EventCategoryInput)
```
 I had to overload the bitwise operators of the enum class **EventCategory**. So, I created a new **Event.cpp** file inside **Hazel/src/Events/** and implemented the overloading of the bitwise operators there.

- [x] I also had to change the virtual function **GetCategoryFlags()** to return an **EventCategory** instead of an int.
also I had to typecast the return value of **IsInCategory()** to a bool.

- [x] Finally, you will see a lot of changes to files: **ApplicationEvent.h**, **KeyEvent.h**, **MouseEvent.h** where I basically added the scoped namespace **EventCategory::** to all the **EVENT_CLASS_CATEGORY** macros. Including the **ImGuiLayer.cpp** where the EventCategory is being used.

The file **Hazelnut/imgui.ini** changed on its own when I run the application to test it. I expect this is expected
behavior. I did not mess with that file at all!
 

Here you can see how **overloading enum class bitwise operations** work:
https://blog.bitwigglers.org/using-enum-classes-as-type-safe-bitmasks/

Thank you for your time!

### CODE REVIEW

**Review by 3x3y3z3t — CHANGES_REQUESTED** [review]

File: Hazel/src/Hazel/Events/ApplicationEvent.h [hunk 1]
```diff
@@ -21,7 +21,7 @@ namespace Hazel {
 		}
 
 		EVENT_CLASS_TYPE(WindowResize)
-		EVENT_CLASS_CATEGORY(EventCategoryApplication)
+		EVENT_CLASS_CATEGORY(EventCategory::EventCategoryApplication)
```
[the remark this answers is not in the document]
Comment (babaliaris) [reply Hazel/src/Hazel/Events/ApplicationEvent.h:24]: Yeap, I thought of doing that :P If you want you can do it and send me a pull request to get the credits.

**Comment by babaliaris** [comment]
I just resolved some **imgui.ini** file conflicts with Cherno's new commit to the master branch. Now the **imgui.ini** is completely clear from changes from this pull request!

**Comment by babaliaris** [comment]
Don't really worry about that imgui.ini commits. I tried to make the changes disappear but for some reason, GitHub likes to add a new line at the end of the imgui.ini file while Cherno's current commit does not contain that new line.

**Comment by TheCherno** [comment]
We should just keep using enums for this case 👍

**Comment by sliptime** [comment]
If the warning still feels like an annoying kid mocking you for not having gotten a complete 0 out of 0 Warnings, you can place: 
```
#pragma warning(disable : 26812)
```
at the top of Event.h, it will disable the enum class warning.

**Pull request closed without merging**